pkg/apis/navigator/v1alpha1/types.go

```diff
@@ -43,6 +43,14 @@ type CassandraClusterNodePool struct {
 	// Persistence specifies the configuration for persistent data for this
 	// node.
 	Persistence PersistenceConfig `json:"persistence,omitempty"`
+
+	// NodeSelector should be specified to force nodes in this pool to run on
+	// nodes matching the given selector.
+	// +optional
+	NodeSelector map[string]string `json:"nodeSelector"`
+
+	Rack       string `json:"rack"`
+	Datacenter string `json:"datacenter"`
```

Comment:
Needs a comment to explain that if these aren't set, a default rack and datacenter will be specified.
Needs +optional on the fields.

Reply:
Also, are we going to auto-populate the `zone` or `datacenter` fields if the node selector includes a selector on zone?

If so, we need to add that functionality and a comment here explaining as such.

Reply:
I think it'd be better to auto-populate the zone based on the nodepool name (as you mentioned below), and the region as some static default.